### ISSUE TITLE
Break mod into sub-sections based on compatibility

### ISSUE THREAD

**Issue by kevinkidwell** [kevinkidwell/customspouseroomlocations#7]
Following a suggestion from dyanosis, going to look into breaking this mod into a main version and sub-files with [subkeys](https://stardewvalleywiki.com/Modding:Modder_Guide/APIs/Update_checks ) to support compatibility.

For some more information, can go to the [Discussion Post](https://www.nexusmods.com/stardewvalley/mods/21525/?tab=posts&jump_to_comment=136436391) with the suggestion, or see quoted below

---

Small suggestion to help those who, like yourself, don't use all of the other mods that this has compatibility for: 

[subkeys ](https://stardewvalleywiki.com/Modding:Modder_Guide/APIs/Update_checks)- Split your mod into the following:
    Vanilla changes - only changes needed for SDV 1.6 and nothing else - This is your Main file
    LNH's First Farmhouse - only changes for this mod - This is an Optional file
    Seasonal Garden Farmhouse v2 - only changes for this mod - This is an Optional file
    Lune Farmhouse - only changes for this mod - This is an Optional file
    Aimon's Tidy Cozy Farmhouse - only changes for this mod - This is an Optional file
    Aimon's Fancy Farmhouse - only changes for this mod - This is an Optional file

And then for each mod that isn't your primary, you simply add "@<some_key_that_makes_sense>". For example, LNH's First Farmhouse could have the subkey "@lnh-ff" (no quotes). So the whole UpdateKeys entry would be:
`"UpdateKeys": ["Nexus:21525@lnh-ff"]﻿`

Your main file (aka with the vanilla changes) would have the entry:
`﻿"UpdateKeys": ["Nexus:21525"]﻿`
Notice that there's no @ since this is your "main" mod.

And then when you post the other Optional (aka compatibility) mods, you just put the appropriate "@<ID>" into the description... like so:
```
﻿Made some changes to improve compatibility.

Manifest Key: @lnh-ff
```

I have mine set up this way for my mod [More Giant Crop Locations](https://www.nexusmods.com/stardewvalley/mods/22815?tab=files). This stops people from having to specify in the config which mod they have and which they don't. It also allows you to make each of the compatibility mods require its specific mod to help alert users when they are using something that they don't actually have. It'll be a much better experience for people.

And yes, they'd have to download more files if they have multiple mods (not sure that's really an issue here as each of the mods you list is changing the farmhouse), but still. With this set up, those of us who just want vanilla would only get vanilla and not a super bloated config that's trying to handle everything.﻿

Also, this stops people from getting notifications that "one of your mods needs updating" because you changed the version and added more compatibility for something they probably aren't using. Similar to how those mods that have translations often have whole new mod pages or have the translations as separate files. So annoying to get updates that "I need to update my mod" when I don't need translation files.

**Comment by Dyanosis** [comment]
I could help if you want. I won't intrude on your turf/space though if you want to do this yourself. I just really want to see others succeeding and getting better! :D

**Comment by kevinkidwell** [comment]
> I could help if you want. I won't intrude on your turf/space though if you want to do this yourself. I just really want to see others succeeding and getting better! :D

Thank you! I totally don't mind the help although I may want to take the initial crack at it, just because it's something new to learn haha. I love community coding and open source projects so I'm not territorial about this stuff because it's for the SDV community!

I'll toss in a PR with that update whenever I get it going, probably sometime this week!

**Comment by kevinkidwell** [comment]
Completed in #8 and released in 2.0.0
